frontend/src/scenes/error-tracking/queries.ts

```diff
@@ -1,47 +1,130 @@
+        const { labels, data } = generateSparklineProps({ value, displayAs, offsetHours })
+
+        select.splice(2, 0, `<Sparkline data={${data}} labels={[${labels.join(',')}]} /> as "context.columns.volume"`)
+        columns.splice(2, 0, 'context.columns.volume')
+    }
+
     return {
         kind: NodeKind.DataTableNode,
         source: {
             kind: NodeKind.EventsQuery,
-            select: [
-                'any(properties) -- Error',
-                'properties.$exception_type',
-                `sparkline(reverse(arrayMap(x -> countEqual(groupArray(dateDiff('hour', now() - INTERVAL 1 day, timestamp)), x), range(24)))) -- Volume`,
-                'count() as unique_occurrences -- Occurrences',
-                'count(distinct $session_id) as unique_sessions -- Sessions',
-                'count(distinct distinct_id) as unique_users -- Users',
-                'max(timestamp) as last_seen',
-                'min(timestamp) as first_seen',
-            ],
+            select: select,
             orderBy: [order],
             ...defaultProperties({ dateRange, filterTestAccounts, filterGroup }),
         },
-        hiddenColumns: [
-            'properties.$exception_type',
-            'first_value(properties)',
-            'max(timestamp) as last_seen',
-            'min(timestamp) as first_seen',
-        ],
+        hiddenColumns: ['properties.$exception_type', 'last_seen', 'first_seen'],
         showActions: false,
         showTimings: false,
+        columns: columns,
     }
 }
 
+export const parseSparklineSelection = (selection: string): SparklineConfig => {
+    if (selection in SPARKLINE_CONFIGURATIONS) {
+        return SPARKLINE_CONFIGURATIONS[selection]
+    }
+
+    const result = selection.match(/\d+|\D+/g)
+
+    if (result) {
+        const [value, unit] = result
+        if (unit === 'y') {
+            return { value: Number(value) * 12, displayAs: 'month' }
+        }
+        return {
+            value: Number(value),
+            displayAs: unit === 'h' ? 'hour' : unit === 'd' ? 'day' : unit === 'w' ? 'week' : 'month',
+        }
+    }
+    return { value: 24, displayAs: 'hour' }
+}
+
+export const generateSparklineProps = ({
+    value,
+    displayAs,
+    offsetHours,
+}: SparklineConfig): { labels: string[]; data: string } => {
+    const offset = offsetHours ?? 0
+    const now = dayjs().subtract(offset, 'hours').startOf(displayAs)
+    const dates = range(value).map((idx) => now.subtract(value - (idx + 1), displayAs))
+    const labels = dates.map((d) => `'${d.format('D MMM, YYYY HH:mm')} (UTC)'`)
+
+    const toStartOfIntervalFn =
+        displayAs === 'minute'
+            ? 'toStartOfMinute'
+            : displayAs === 'hour'
+            ? 'toStartOfHour'
+            : displayAs === 'day'
+            ? 'toStartOfDay'
+            : displayAs === 'week'
+            ? 'toStartOfWeek'
+            : 'toStartOfMonth'
```

Comment:
similar to above could this be a map with a defaiult, nicer to read than this chain 😊